### ISSUE TITLE
Instagram Block URL Signature Expired

### ISSUE THREAD

**Issue by rsmith4321** [Automattic/jetpack#22183]
I noticed that sometimes the images in an Instagram block with not display, if I try to view the img url it says "URL Signature Expired".  This is obviously an issue with page caching.  Instagram expires the images after a certain period of time. Would there be any way to use the Instagram block and full page caching at the same time? Perhaps I just need a shorter page cache time.  Any suggestions?

**Comment by loyukfai** [comment]
Yes, it happens. I just set a shorter page cache time. How long is yours?

**Comment by rsmith4321** [comment]
I just switched to the Smash Balloon Instagram feed plugin. It caches the images locally and also resizes them. It seems to be the only solution to the issue.  I use Cloudflare APO so I can't really control the page cache time. I think it would be a good idea for Jetpack to add a warning that the Instagram block is incompatible with Cloudflare APO and page cache over a certain time. Or if they could also implement local caching.
